torchchat/cli/builder.py

```diff
@@ -294,12 +304,13 @@ def validate_model(
         if model is None:
             return
 
-        if sum([self.is_tiktoken, self.is_hf_tokenizer, self.is_sentencepiece]) != 1:
+        if self.is_tokenizer_none():
```

Comment:
Let's keep it simple and check for `self.tokenizer_type == TokenizerType.NONE` here

We can skip the the assert/helper